### PR TITLE
Add --tail to stream create and kafka-provider create

### LINKED ISSUE #134: Add `--tail` and `--wait-timeout` flags for `*-provider create` and `stream create`
It's useful to know if a provider or stream was created successfully. While streams don't have logs to tail, we should use the same flags for consistency with other commands.

In particular, since creating a processor requires stream to be ready, it's helpful to know in advance that the streams are indeed ready, or not.

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #134

### CODE REVIEW

File: pkg/streaming/commands/stream_create.go [hunk 1]
```diff
@@ -20,6 +20,10 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
+
+	"github.com/projectriff/cli/pkg/k8s"
+	"github.com/projectriff/cli/pkg/race"
```
Comment (scothis) [inline pkg/streaming/commands/stream_create.go]: rm trailing line break

File: pkg/streaming/commands/kafkaprovider_create.go [hunk 2]
```diff
@@ -20,6 +20,10 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
+
+	"github.com/projectriff/cli/pkg/k8s"
+	"github.com/projectriff/cli/pkg/race"
```
Comment (scothis) [inline pkg/streaming/commands/kafkaprovider_create.go]: rm trailing line break

File: pkg/streaming/commands/kafkaprovider_create.go [hunk 3]
```diff
@@ -75,6 +88,23 @@ func (opts *KafkaProviderCreateOptions) Exec(ctx context.Context, c *cli.Config)
 		}
 	}
 	c.Successf("Created kafka provider %q\n", provider.Name)
+	if opts.Tail {
+		err := race.Run(ctx, opts.WaitTimeout,
```
Comment (scothis) [inline pkg/streaming/commands/kafkaprovider_create.go]: do we want to collect logs from the provisioner and gateway deployments as well?

File: pkg/streaming/commands/kafkaprovider_create_test.go [hunk 4]
```diff
@@ -55,6 +56,25 @@ func TestKafkaProviderCreateOptions(t *testing.T) {
...
+		},
+		{
+			Name: "invalid timeout",
+			Options: &commands.KafkaProviderCreateOptions{
+				ResourceOptions:  rifftesting.ValidResourceOptions,
+				BootstrapServers: "localhost:9092",
+				WaitTimeout:      -4 * time.Second,
+			},
+			ExpectFieldErrors: cli.ErrInvalidValue(-4*time.Second, cli.WaitTimeoutFlagName),
+		},
```
Comment (scothis) [inline pkg/streaming/commands/kafkaprovider_create_test.go]: add tests for the command execution as well

File: pkg/streaming/commands/stream_create_test.go [hunk 5]
```diff
@@ -80,6 +81,25 @@ func TestStreamCreateOptions(t *testing.T) {
...
+		},
+		{
+			Name: "negative timeout",
+			Options: &commands.StreamCreateOptions{
+				ResourceOptions: rifftesting.ValidResourceOptions,
+				Provider:        "test-provider",
+				WaitTimeout:     -3 * time.Second,
+			},
+			ExpectFieldErrors: cli.ErrInvalidValue(-3*time.Second, cli.WaitTimeoutFlagName),
+		},
```
Comment (scothis) [inline pkg/streaming/commands/stream_create_test.go]: add tests for the command execution as well

**Comment by ericbottard** [comment]
@scothis  Please take another look.

File: pkg/streaming/commands/stream_create_test.go [hunk 6]
```diff
@@ -202,6 +229,80 @@ Created stream "my-stream"
...
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: defaultNamespace,
+						Name:      "input",
+					},
+					Spec: streamv1alpha1.StreamSpec{
+						ContentType: "application/json",
+						Provider:    "franz",
+					},
+				},
+			},
```
Comment (scothis) [inline pkg/streaming/commands/stream_create_test.go]: include ExpectOutput

File: pkg/streaming/commands/stream_create_test.go [hunk 7]
```diff
@@ -202,6 +229,80 @@ Created stream "my-stream"
...
+			ShouldError: true,
```
Comment (scothis) [inline pkg/streaming/commands/stream_create_test.go]: include ExpectOutput

**Pull request merged**